### DISCUSSION TITLE
Make Right Mouse Button Behavior Consistent with the Platform Being Used

### DISCUSSION THREAD

**Discussion by jtwine** [godotengine/godot-proposals#7463] [Editor]
So this discussion comes out of a [Godot Issue](https://github.com/godotengine/godot/issues/80346) I reported on how the behavior of the Right Mouse Button (RMB), when using Godot on Windows, is inconsistent with normal/standard Windows behavior.  

[Sauermann](https://github.com/Sauermann) correctly indicated that the RMB behavior is different on different platforms, and so this discussion on the best way to abstract the behavior of the RMB (and possibly other UI elements/actions/peripherals) so that when used on a particular platform, correct and expected behavior results.

**Comment by lostminds** [comment]
This sounds like the right approach to me, and a good ambition to have for both the editor and maybe default behavior for control nodes as well.

On macOS native menus generally support selection using two-click (click once to open, click again to select) or click-drag-release (press down to open menu, drag mouse and then release to select item). They also generally work using either the right mouse button or the left mouse button (but not middle mouse). In other words you can use either of the four possible combinations of right or left button, two-click or click-drag-release in menus. Many apps also support using control-left-click as an alternative to right-click for context menus as a memory from the old one mouse button days.

Testing in the 4.2.1 Godot editor on macOS, the Godot editor menus currently only seem to support two-click left mouse button selection, so there's potential there for supporting three more ways of selecting menu items some user might prefer on macOS.
